frontend/src/container/Sidebar.jsx

```diff
@@ -128,41 +127,47 @@ export class Sidebar extends React.Component {
     return !unLoggedAllowedPageList.some(url => props.location.pathname.startsWith(url))
   }
 
+  hasClass = (element, className) => {
+    const hasOriginalClass = element.classList.contains(className)
+    const hasStopClass = element.classList.contains('sidebar')
+    if (hasStopClass) return false
+    else return hasOriginalClass || this.hasClass(element.parentNode, className)
+  }
+
   handleClickToggleSidebar = (e) => {
-    e.stopPropagation()
-    GLOBAL_dispatchEvent({
-      type: this.state.isSidebarClosed
-        ? CUSTOM_EVENT.SHOW_SIDEBAR
-        : CUSTOM_EVENT.HIDE_SIDEBAR,
-      data: {}
-    })
-    this.setState(previousState => ({ isSidebarClosed: !previousState.isSidebarClosed }))
+    const { state } = this
+    const isExpandOrEmptyZone = this.hasClass(e.target, 'sidebar__header__expand') || this.hasClass(e.target, 'sidebar__emptyZone')
+    const hasLockClass = this.hasClass(e.target, 'lockToggleSidebarWhenOpenedOnMobile')
```

Comment:
use a constant for lockToggleSidebarWhenOpenedOnMobile to avoid typo and to show it is specific intelligence not related to css

Reply:
fixed